sixtyfps_runtime/corelib/sharedvector.rs

```diff
@@ -251,6 +245,32 @@ impl<T: Clone> SharedVector<T> {
             }
         }
     }
+
+    fn shrink(&mut self, new_len: usize) {
+        if self.len() == new_len {
+            return;
+        }
+
+        debug_assert!(
+            unsafe { self.inner.as_ref().header.refcount.load(atomic::Ordering::Relaxed) } == 1
+        );
+        // Safety: caller (and above debug_assert) must ensure that the array is not shared.
+        let mut inner = unsafe { self.inner.as_mut() };
+
+        while inner.header.size > new_len {
+            inner.header.size -= 1;
+            // Safety: The array was of size inner.header.size, so there should be an element there
+            unsafe {
+                core::ptr::drop_in_place(inner.data.as_mut_ptr().add(inner.header.size));
+            }
+        }
+    }
+
+    /// Clears the vector and removes all elements. The capacity remains unaffected.
+    pub fn clear(&mut self) {
+        self.detach(self.capacity());
```

Comment:
But detach will do a deep clone of everything inside, we don't want that when we want to clear. That's quite unintuitive.